log/sequencer.go

```diff
@@ -209,32 +209,29 @@ func (s Sequencer) prepareLeaves(leaves []*trillian.LogLeaf, begin int64, label
 
 func (s Sequencer) updateCompactTree(mt *compact.Tree, leaves []*trillian.LogLeaf, label string) (map[string]storage.Node, error) {
 	nodeMap := make(map[string]storage.Node)
+	store := func(level int, index int64, hash []byte) error {
+		id, err := storage.NewNodeIDForTreeCoords(int64(level), index, maxTreeDepth)
+		if err != nil {
+			return err
+		}
+		nodeMap[id.String()] = storage.Node{NodeID: id, Hash: hash}
+		return nil
+	}
+
 	// Update the tree state by integrating the leaves one by one.
 	for _, leaf := range leaves {
 		seq, err := mt.AddLeafHash(leaf.MerkleLeafHash, func(depth int, index int64, hash []byte) error {
-			nodeID, err := storage.NewNodeIDForTreeCoords(int64(depth), index, maxTreeDepth)
-			if err != nil {
-				return err
-			}
-			nodeMap[nodeID.String()] = storage.Node{
-				NodeID: nodeID,
-				Hash:   hash,
-			}
-			return nil
+			return store(depth, index, hash)
```

Comment:
Can you not just pass `store` directly?

Reply:
Indeed, will do.

Reply:
cancel(): In a follow-up commit store() no longer returns an error (it could, but then it would require handling it in 2 other places).